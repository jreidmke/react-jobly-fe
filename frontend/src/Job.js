import { useContext, useState, useEffect } from "react";
import { Link } from 'react-router-dom';
import UserContext from "./UserContext";

const Job = ({company, title, salary, equity, id}) => {
    const {hasAppliedToJob, apply} = useContext(UserContext);
    const [applied, setApplied] = useState();

    useEffect(function updateApplication() {
        setApplied(hasAppliedToJob(id));
    }, [id, hasAppliedToJob]); 

    async function handleApply(e) {
        e.preventDefault();
        if(hasAppliedToJob(id)) return;
        apply(id);
        setApplied(true);
    }

    return(
        <div style={{width: '50%', margin: 'auto'}}>
            <h1>{company}</h1>
            <h3>{title}</h3>
            <button onClick={handleApply} disabled={applied}>{applied ? "Applied" : "Apply"}</button>
            <p>Salary: {salary}</p>
            <p>Equity: {equity}</p>
            <hr></hr>
        </div>
    )
}

export default Job;