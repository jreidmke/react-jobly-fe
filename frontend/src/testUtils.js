import React from "react";
import UserContext from "./UserContext";

const demoUser = {username: "jreid", firstName: "James", lastName: "Reid", email: "jreidmke@gmail.com", isAdmin: false, applcations: []};

const UserProvider =
    ({ children, currUser = demoUser, hasAppliedToJob = () => false }) => (
    <UserContext.Provider value={{ currUser, hasAppliedToJob }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
