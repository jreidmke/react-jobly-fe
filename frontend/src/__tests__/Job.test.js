import React from "react";
import { render } from "@testing-library/react";
import Job from "../Job";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";

it("renders without crashing", () => {
    render(
        <UserProvider>
            <MemoryRouter>
                <Job
                company="Pizza Co."
                title="pizza chef"
                salary={450}
                equity={.1}
                 />
            </MemoryRouter>
        </UserProvider>        
    )
});

it("matches the snapshot", () => {
    const {asFragment} = render(
        <UserProvider>
            <MemoryRouter>
                <Job
                company="Pizza Co."
                title="pizza chef"
                salary={450}
                equity={.1}
                 />
            </MemoryRouter>
        </UserProvider>          
    );
    expect(asFragment()).toMatchSnapshot();
});