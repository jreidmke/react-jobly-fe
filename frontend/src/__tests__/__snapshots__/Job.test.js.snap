// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 50%; margin: auto;"
  >
    <h1>
      Pizza Co.
    </h1>
    <h3>
      pizza chef
    </h3>
    <button>
      Apply
    </button>
    <p>
      Salary: 450
    </p>
    <p>
      Equity: 0.1
    </p>
    <hr />
  </div>
</DocumentFragment>
`;
